addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -1087,10 +1063,21 @@ public void run() {
                                 vehicle = queryVehicle();
                             }
                         }
+                    } else {
+                        Thread.sleep(250);
+                    }
+                } catch (IOException | InterruptedException e) {
```

Comment:
Thanks, looks better.
One other thing just caught my eye: Upon handler disposal, you are interrupting the thread, so an InterruptedException is absolutely normal and expected here. Thus, there is no reason to taint the log file with an error message, right?